sql-plugin/src/main/spark320/scala/org/apache/spark/sql/rapids/execution/python/shims/GpuArrowPythonRunner.scala

```diff
@@ -88,6 +89,15 @@ class GpuArrowPythonRunner(
       val isInputNonEmpty = inputIterator.nonEmpty
       lazy val arrowSchema = ArrowUtilsShim.toArrowSchema(pythonInSchema, timeZoneId)
 
+      override def run(): Unit = Utils.logUncaughtExceptions {
+        TaskContext.setTaskContext(context)
+        // For writer thread, it is essentially a split of the original task, so we want to
+        // treat it exactly like the main task thread, e.g. being registered for its whole lifecycle
+        TaskRegistryTracker.registerThreadForRetry()
+        RmmSpark.bindPropagateThreads(RmmSpark.getCurrentThreadId, readerNativeThreadId)
```

Comment:
What happens if the user disables GPU python acceleration? We will still end up with two threads for each task, but they are not going to be tied to each other. I would rather have us special case the thread pools, because they are not likely to happen in vanilla apache spark, instead of trying to special case the python UDFs.

Reply:
good point. I'll change acccordingly

Reply:
I think we want to do it in all the places so that we don't miss one on accident.

Reply:
I think we want to do it in all the places so that we don't miss one on accident.

Reply:
it's easy to add everywhere, but it's not easy to know when to remove.  So eventually we'll have to resort to onTaskCompletion() to do the unregisteration?